src/librustdoc/html/static/main.js

```diff
@@ -649,8 +649,7 @@ function defocusSearchBar() {
          */
         function execQuery(query, searchWords, filterCrates) {
             function itemTypeFromName(typename) {
-                var length = itemTypes.length;
-                for (var i = 0; i < length; ++i) {
```

Comment:
Why does this change anything? The only difference is whether `len` is declared in the loop initializer, but the initializer only runs once, not each iteration of the loop.

Anyway, this is better written with the standard library:

```suggestion
var index = itemTypes.indexOf(typename);
if (index === -1) return NO_TYPE_FILTER;
return index;
```

Reply:
It's maybe "better looking" but it's slower apparently (8% slower). I used this code to test:

```js
function foo(itemTypes, typename) {
	return itemTypes.indexOf(typename);
}
```

```js
function foo(itemTypes, typename) {
    for (var i = 0, len = itemTypes.length; i < len; ++i) {
	if (itemTypes[i] === typename) {
	     return i;
	}
    }
    return -1;
}
```

And then run the `foo` function in:

```js
var x = ["a", "akjsdfsdfs", "sdfqdfdqfqd", "kjsdbfqjkfqd", "qdkjfqdkjfqdfjkqsdfq", "qskjdfkqjsdb", "aizgeiazuedaz", "azlk,lcx,q", "qdsfns"];
for (var i = 0; i < 100000; ++i) {
	foo(x, "qdsfns");
}
```

(sill on https://jsbench.me/)

Reply:
Huh, weird. Well it's readable enough as-is so fine to keep it this way.

Reply:
Welcome to the JS world. How do you like it so far? :p